files/default/ndb_templates/ndb_config.ini

```diff
@@ -0,0 +1,212 @@
+LongMessageBuffer=64M
+
+MaxFKBuildBatchSize=64
+MaxReorgBuildBatchSize=64
+EnablePartialLcp=1
+InsertRecoveryWork=40
+
+#The maximum time in ms that is permitted to lapse between operations in the same transaction before the transaction is aborted.
+TransactionInactiveTimeout=1500
+
+TransactionDeadlockDetectionTimeout=1500
+
+#0: Disables locking. This is the default value.
+#1: Performs the lock after allocating memory for the process.
+#2: Performs the lock before memory for the process is allocated.
+# Needs root privileges
+LockPagesInMainMemory=1
+
+# Setting these parameters allows you to take advantage of real-time scheduling
+# of NDBCLUSTER threads to get higher throughput.
+RealTimeScheduler=0
+
+# Savings of up to 50% over noncompressed LCPs and backups
+CompressedLCP=0
+CompressedBackup=1
+
+# Deprecated from version 5.6.7
+#BackupMaxWriteSize= 1M
+#BackupDataBufferSize= 16M  
+
+
+BackupLogBufferSize= 16M
+
+#The maximum size of the memory unit to use when allocating memory for tables
+MaxAllocate=32M
+
+#The size of the table hash maps used by NDB
+DefaultHashMapSize=3840
+
+# Needs root privileges
+ODirect=0
+
+# total amount of memory (in bytes) to be allocated by each node for which it is set for use among all configured transporters for all types of nodes (NDB, API, etc)
+TotalSendBufferMemory=16M
+# amount of transporter send buffer memory to allocate in addition to any that has been set
+ExtraSendBufferMemory=0
+
+# This parameter specifies the time in microseconds for threads to be executed in the scheduler before sleeping.
+# Default = 0. Higher value to optimize latency over throughput.
+SchedulerSpinTimer=0 
+
+# Specifies the time in microseconds for threads to be executed in the scheduler before being sent. 
+# Default is 50. Higher values give higher throughput at cost of increased latency.
+SchedulerExecutionTimer=50            
+
+TwoPassInitialNodeRestartCopy=true
+# Number of threads to create when rebuilding indexes during a system or node start
+BuildIndexThreads=128
+
+# Disable for 7.5.+
+#SchedulerResponsiveness=5
+
+Numa=1
+
+# On-Disk data column configuration
+DiskPageBufferEntries=10
+DiskPageBufferMemory=512M
+# Memory used by disk data for buffers/paging
+SharedGlobalMemory=512M
+# number of unbound threads used for Disk Data file access
+DiskIOThreadPool=8
+
+
+# Move this to another drive if you have a high number of ops/sec
+InitialLogFileGroup=undo_buffer_size=128M;lg1.dat:4G
+
+# Move this to another drive to store small files in HopsFS
+InitialTablespace=name=ts_1;extent_size=16M;ts1.dat:8G
+
+# https://mikaelronstrom.blogspot.com/2018/08/more-automated-control-in-mysql-cluster.html
+EnableRedoControl=1
+
+
+
+
+[MYSQLD DEFAULT]
+
+[NDB_MGMD DEFAULT]
+
+[TCP DEFAULT]
+OverloadLimit=0
+
+#
+# Supports up to Max 3 ndb_mgmds. First one gets ArbitrationRank=1, others gets lower priority ArbitrationRank=2
+#
+{{range .Mgmd}}
```

Comment:
I guess you template this file in ec2-init, right? 
Isn't better to generate these files instead of templating?

Reply:
This is templated by `ndb-agent` It's a daemon heartbeating to hopsworks.ai We template it because some configuration properties will not change.